### PR TITLE
lsusb-t: Emit USB IDs and other handy info when verbosity is increased

### PR DESCRIPTION
**Pull request by dlitz**
TL;DR: 
```
$ ./lsusb -t | head -3
/:  Bus 04.Port 1: Dev 1, Class=root_hub, Driver=xhci_hcd/2p, 5000M
/:  Bus 03.Port 1: Dev 1, Class=root_hub, Driver=ehci-pci/3p, 480M
    |__ Port 1: Dev 2, If 0, Class=Hub, Driver=hub/8p, 480M

$ ./lsusb -tv | head -6    # new!
/:  Bus 04.Port 1: Dev 1, Class=root_hub, Driver=xhci_hcd/2p, 5000M
    ID 1d6b:0003 Linux Foundation 3.0 root hub
/:  Bus 03.Port 1: Dev 1, Class=root_hub, Driver=ehci-pci/3p, 480M
    ID 1d6b:0002 Linux Foundation 2.0 root hub
    |__ Port 1: Dev 2, If 0, Class=Hub, Driver=hub/8p, 480M
        ID 8087:0024 Intel Corp. Integrated Rate Matching Hub

$ ./lsusb -tvv | head -9    # new!
/:  Bus 04.Port 1: Dev 1, Class=root_hub, Driver=xhci_hcd/2p, 5000M
    ID 1d6b:0003 Linux Foundation 3.0 root hub
    /sys/bus/usb/devices/usb4  /dev/bus/usb/004/001
/:  Bus 03.Port 1: Dev 1, Class=root_hub, Driver=ehci-pci/3p, 480M
    ID 1d6b:0002 Linux Foundation 2.0 root hub
    /sys/bus/usb/devices/usb3  /dev/bus/usb/003/001
    |__ Port 1: Dev 2, If 0, Class=Hub, Driver=hub/8p, 480M
        ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
        /sys/bus/usb/devices/3-1  /dev/bus/usb/003/002
```

lspci (from pciutils) has the ability to show bus topology and vendor/product ID in the same output (e.g. `lspci -tvnn`).  This PR attempts to bring something similar to lsusb (albeit somewhat constrained by the existing output format of `lsusb --tree`).

Motivation:  I have some flaky hardware that sometimes enumerates USB 3.0-capable hardware under ehci instead of xhci.  I sometimes also care about which devices share a common bus, such when running virtualized with PCI passthrough (e.g. Qubes).

This is intended as an RFC, but feel free to merge it as-is if it suits you.  The specific output format and the choice to output /sys and /dev paths when verbosity >= 2 were more-or-less arbitrary.  The main thing I'd like is getting vendor/product info from lsusb's --tree mode.

### CODE REVIEW

**Comment by gregkh** [comment]
Looks like a nice feature change, thanks for this!

Does any documentation also need to be fixed up for this new feature so that people have a chance to find it?
